packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -110,23 +110,11 @@ export class PendingStateManager implements IDisposable {
 		);
 		if (!this.pendingMessages.isEmpty()) {
 			return {
-				pendingStates: this.pendingMessages.toArray().map((message) => {
-					// ! TODO: Remove conversion to IPendingMessageOld in "2.0.0-internal.6.0.0" AB#3826
-					const content = JSON.parse(message.content);
-					// IdAllocations need their localOpMetadata stashed in the contents
-					// of the op to correctly resume the session when processing stashed ops
-					if (content.type === ContainerMessageType.IdAllocation) {
-						content.contents.stashedState = message.localOpMetadata;
```

Comment:
@justus-camp Is it possible to have this `stashedState` be part of the content when submitting the message (i.e. in `maybeSubmitIdAllocationOp`)?
I want to avoid `parse` then `serialize` as much as possible just to check the type add another prop to the content string

Reply:
I used localOpMetadata because we don't want to submit the `stashedState` over the wire but need it to rehydrate the IdCompressor for stashed ops. My initial prototype actually did what you're suggesting and deleted `stashedState` from the op before submission, but there's [a check](https://github.com/microsoft/FluidFramework/blob/main/packages/runtime/container-runtime/src/pendingStateManager.ts#L241C21-L241C21) in `PendingStateManager::processPendingLocalMessage` that makes doing it this way difficult and causes a `DataProcessingError`. I suppose we could ignore `IdAllocation` ops in that check but that's pretty unfortunate. 

It's worth noting that overall this stashed op implementation isn't great and we kind of "retro-fitted" the original compressor to do it this way.

Reply:
Is it possible to move the serialized state to a summary blob? I think it should, but I may be missing something. If so, we can leave logic in the `PendingStateManager` for back/forward compat while that goes on.

If not possible, I'll add some logic in `ContainerRuntime` to handle this conversion of content for the `PendingStateManager`. The PSM shouldn't know about specific types of ops if avoidable.